dns/asyncresolver.py

```diff
@@ -394,3 +394,76 @@ async def zone_for_name(
             name = name.parent()
         except dns.name.NoParent:  # pragma: no cover
             raise NoRootSOA
+
+
+async def make_resolver_at(
+    where: Union[dns.name.Name, str],
+    port: int = 53,
+    family: int = socket.AF_UNSPEC,
+) -> Resolver:
+    """Make a stub resolver using the specified destination as the full resolver.
+
+    *where*, a ``dns.name.Name`` or ``str`` the domain name or IP address of the
+    full resolver.
+
+    *port*, an ``int``, the port to use.  If not specified, the default is 53.
+
+    *family*, an ``int``, the address family to use.  This parameter is used if
+    *where* is not an address.  The default is ``socket.AF_UNSPEC`` in which case
+    the first address returned by ``resolve_name()`` will be used, otherwise the
+    first address of the specified family will be used.
+
+    Returns a ``dns.resolver.Resolver`` or raises an exception.
+    """
+    if isinstance(where, str) and dns.inet.is_address(where):
+        address = where
+    else:
+        answers = await resolve_name(where, family)
+        address = list(answers.addresses())[0]
```

Comment:
Shouldn't this be building a list of all of the addresses, not just the first one?

Reply:
Probably!  I was worried about doubling up on errors when the same server had a v4 and v6 address, say.  But of course you don't know if servers are the "same" or not, so I should take them all.